### PR TITLE
Implemented AppId class with login function

### PR DESCRIPTION
**Pull request by dylancanfield**
Added a function to authenticate against the App ID backend

### CODE REVIEW

File: lib/vault/api/app_id.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    # Create an authentication token.
+    #
+    # @example
+    #   Vault.app_id.login('foo', 'bar') #=> #<Vault::Secret lease_id="">
+    #
+    # @param [String] app_id
+    # @param [String] user_id
+    #
+    # @return [Secret]
+    def login(app_id, user_id)
```
Comment (sethvargo) [inline lib/vault/api/app_id.rb]: I think, as part of this request, we want to save the resulting secret back onto the system in `$HOME/.vault-token` and then also set the Client's .token parameter to the returned value.

Comment (sethvargo) [inline lib/vault/api/app_id.rb]: Actually, after a bit more consideration, it _shouldn't_ persist to disk, but it should store the resulting token in memory for future requests I think.

**Comment by dylancanfield** [comment]
OK, client's token attribute is now updated for future requests.

**Comment by sethvargo** [comment]
Cleaned up and added more login endpoints in https://github.com/hashicorp/vault-ruby/pull/26. Thanks!

**Pull request closed without merging**